plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,188 @@
+		$info            = array(
+			'status_code' => $status_code,
+			'available'   => false,
+		);
+
+		if (
+			400 === $status_code
+			&& isset( $data['data']['params'] )
+			&& is_array( $data['data']['params'] )
+			&& count( $expected_params ) === count( array_intersect( $data['data']['params'], $expected_params ) )
+		) {
+			// The REST API endpoint is available.
+			$info['available'] = true;
+		} elseif ( 401 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Authorization should not be required to access the REST API endpoint.', 'optimization-detective' );
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				esc_html__( 'To collect URL metrics, the REST API endpoint should be accessible without requiring authorization.', 'optimization-detective' )
+			);
+		} elseif ( 403 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'The REST API endpoint should not be forbidden.', 'optimization-detective' );
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				esc_html__( 'The REST API endpoint is blocked. Please review your server or security settings.', 'optimization-detective' )
+			);
+		} else {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Error accessing the REST API endpoint', 'optimization-detective' );
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				esc_html__( 'There was an issue reaching the REST API endpoint. This might be due to server settings or the REST API being disabled.', 'optimization-detective' )
+			);
+		}
+		$info['error_message'] = $result['label'];
+	}
+
+	update_option( 'od_rest_api_info', $info );
+	return $result;
+}
+
+/**
+ * Periodically runs the Optimization Detective REST API health check.
+ *
+ * @since n.e.x.t
+ */
+function od_schedule_rest_api_health_check(): void {
+	if ( ! (bool) wp_next_scheduled( 'od_rest_api_health_check_event' ) ) {
+		wp_schedule_event( time(), 'weekly', 'od_rest_api_health_check_event' );
+	}
+}
+
+/**
+ * Hook for the scheduled REST API health check.
+ *
+ * @since n.e.x.t
+ */
+function od_run_scheduled_rest_api_health_check(): void {
+	od_optimization_detective_rest_api_test();
+}
+
+/**
+ * Displays an admin notice if the REST API health check fails.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $plugin_file Plugin file.
+ */
+function od_rest_api_health_check_admin_notice( string $plugin_file ): void {
+	if ( 'optimization-detective/load.php' !== $plugin_file ) {
+		return;
+	}
+
+	$od_rest_api_info = get_option( 'od_rest_api_info', array() );
+	if (
+		isset( $od_rest_api_info['available'] ) &&
+		! (bool) $od_rest_api_info['available'] &&
+		isset( $od_rest_api_info['error_message'] )
+	) {
+		wp_admin_notice(
+			esc_html( $od_rest_api_info['error_message'] ),
+			array(
+				'type'               => 'warning',
+				'additional_classes' => array( 'inline', 'notice-alt' ),
```

Comment:
Related to https://github.com/WordPress/performance/pull/1762/files#r1907962460, I think this should not use `inline` since a user may not scroll down to see the error message with the activated plugin row.

```suggestion
				'additional_classes' => array( 'notice-alt' ),
```